osquery/sql/sql.cpp

```diff
@@ -127,16 +117,23 @@ QueryData SQL::selectFrom(const std::initializer_list<std::string>& columns,
                           ConstraintOperator op,
                           const std::string& expr) {
   PluginRequest request = {{"action", "generate"}};
-  {
-    // Create a fake content, there will be no caching.
-    QueryContext ctx;
-    ctx.constraints[column].add(Constraint(op, expr));
+  // Create a fake content, there will be no caching.
+  QueryContext ctx;
+  ctx.constraints[column].add(Constraint(op, expr));
+  if (columns.size() > 0) {
     ctx.colsUsed = UsedColumns(columns);
-    TablePlugin::setRequestFromContext(ctx, request);
   }
+  TablePlugin::setRequestFromContext(ctx, request);
 
   PluginResponse response;
   Registry::call("table", table, request, response);
+  response.erase(
+      std::remove_if(response.begin(),
+                     response.end(),
+                     [&ctx, &column](const PluginRequest& row) -> bool {
+                       return !ctx.constraints[column].matches(row.at(column));
```

Comment:
As far as I know `std::map::at(...)` throws exception in case of key absence. Is there any guarantee that this key is in the `row`?

Reply:
In code above I'm explicitly adding column to request, so in case if column not present it's clear logic error in table so, In my opinion, the best what we can do here is crash